mrbgems/mruby-objectspace/src/mruby_objectspace.c

```diff
@@ -344,13 +344,38 @@ os_memsize_of(mrb_state *mrb, mrb_value self)
   return mrb_fixnum_value(total);
 }
 
+struct os_memsize_of_all_cb_data {
+  mrb_int t;
+  struct RClass *type;
+};
+
+static int
+os_memsize_of_all_cb(mrb_state *mrb, struct RBasic *obj, void *d)
+{
+  struct os_memsize_of_all_cb_data *data = (struct os_memsize_of_all_cb_data *)d;
+  if(obj->c == data->type) data->t += os_memsize_of_object(mrb, mrb_obj_value(obj));
```

Comment:
You have to use `mrb_class_real()` here since `obj->c` may refer to a singleton class.
```C
if (mrb_class_real(obj->c) == data->type) data->t += os_memsize_of_object(mrb, mrb_obj_value(obj));
```